Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c

```diff
@@ -266,43 +191,35 @@ FwuTopSwapSetting (
     pFwUpdStatus = (FW_UPDATE_STATUS *)(UINTN)RsvdBase;
   }
 
-  //
-  // Get Top swap register Bit0 in PCH Private Configuration Space.
-  //
-  P2sbBase   = MM_PCI_ADDRESS (0, PCI_DEVICE_NUMBER_PCH_LPC, 1, 0); // P2SB device base
-
-  if (MmioRead16 (P2sbBase) == 0xFFFF) {
-    //
-    // unhide P2SB
-    //
-    MmioWrite8 (P2sbBase + 0xE1, 0);
-    DEBUG ((DEBUG_INFO, "P2sb is hidden, unhide it\n"));
-  }
-
-  P2sbBar    = MmioRead32 (P2sbBase + 0x10);
-  P2sbBar  &= 0xFFFFFFF0;
-  ASSERT (P2sbBar != 0xFFFFFFF0);
-
-  TopSwapReg = P2sbBar | ((PID_RTC_HOST) << 16) | (UINT16)(R_RTC_PCR_BUC);
-  Data32    = MmioRead32 (TopSwapReg);
-  DEBUG ((DEBUG_INFO, "TopSwapReg=0x%x\n",  Data32));
   if (pFwUpdStatus->StateMachine == FW_UPDATE_SM_PART_A) {
-    if (GetCurrentBootPartition() == 0) {
-      SetBootPartition(1);
-      ResetSystem(EfiResetCold);
+    if (GetCurrentBootPartition () == PrimaryPartition) {
+      // if on or above failure threshold here, continue to recover the partition
+      if (PcdGetBool (PcdSblResiliencyEnabled) &&
```

Comment:
Realizing we need to cover the disabled case here as well. Will push another change.